cmd/utils.go

```diff
@@ -79,6 +87,87 @@ func parsePageRange(pageRange string) ([]int, error) {
 	return pages, nil
 }
 
+func parseInputPaths(inputPaths []string, recursive bool, matcher fileMatcher) ([]string, error) {
+	var files []string
+	for _, inputPath := range inputPaths {
+		fi, err := os.Stat(inputPath)
+		if err != nil {
+			return nil, err
+		}
+
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			dirFiles, err := parseInputDir(inputPath, recursive, matcher)
+			if err != nil {
+				return nil, err
+			}
+			files = append(files, dirFiles...)
+		case mode.IsRegular():
+			if matcher == nil || matcher(inputPath) {
+				files = append(files, inputPath)
+			}
+		}
+	}
+
+	return files, nil
+}
+
+func parseInputDir(dir string, recursive bool, matcher fileMatcher) ([]string, error) {
+	dirFiles, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	var files []string
+	for _, dirFile := range dirFiles {
+		inputPath := filepath.Join(dir, dirFile.Name())
+
+		switch mode := dirFile.Mode(); {
+		case mode.IsDir():
+			if !recursive {
+				continue
+			}
+
+			subdirFiles, err := parseInputDir(inputPath, recursive, matcher)
```

Comment:
May want to add recursive protection if there is a symlink to a lower directory that could lead to an endless loop.
Can be done by caching the result for parseInputDir and if the result already exists return it back.